### PR TITLE
Add support matrix for Cognitive Services

### PR DESCRIPTION
**Pull request by compulim**
## Description

Update our `SPEECH.md` to include support matrix for Cognitive Services Speech Services.

## Specific Changes

- Add a support matrix table to `SPEECH.md`

---

### CODE REVIEW

File: SPEECH.md [hunk 1]
```diff
@@ -6,6 +6,213 @@ We assume you have already set up a bot and have Web Chat running on a page.
...
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>STT</td>
+      <td>Basic recognition</td>
+      <td><center>✔ 4.6</center></td>
+      <td><center>✔ 4.6</center></td>
+      <td><center>✔ 4.6</center></td>
+      <td><center><nobr>❌ *1</nobr></center></td>
```
Comment (corinagum) [inline SPEECH.md]:
```suggestion
      <td><center><nobr>❌ <sup>*1</sup></nobr></center></td>
```

preferably with an anchor

File: SPEECH.md [hunk 2]
```diff
@@ -6,6 +6,213 @@ We assume you have already set up a bot and have Web Chat running on a page.
...
+      <td><center>✔</center></td>
+      <td><center><nobr>❓ *2</nobr></center></td>
+      <td><center>✔</center></td>
+    </tr>
+  </tbody>
+</table>
+
+Notes:
+
+1. Web View on iOS is not a full browser. It does not have audio recording capabilities, which is required for Cognitive Services
```
Comment (corinagum) [inline SPEECH.md]: add anchors

**Comment by compulim** [comment]
Please review, and be as nitpicking as possible. 😉

File: SPEECH.md [hunk 3]
```diff
@@ -6,6 +6,215 @@ We assume you have already set up a bot and have Web Chat running on a page.
...
+  </thead>
+  <tbody>
+    <tr>
+      <td>STT</td>
+      <td>Basic recognition</td>
+      <th>✔</th><td>4.6</td>
+      <th>✔</th><td>4.6</td>
+      <th>✔</th><td>4.6</td>
+      <th>✔</th><td>4.6</td>
+      <th>❌</th><td><a href="#notes"><sup>*1</sup></a></td>
```
Comment (corinagum) [inline SPEECH.md]: Link to individual numbers? the anchors don't appear to work either, just refreshes the page. I don't know if this is an issue from being a PR

File: SPEECH.md [hunk 4]
```diff
@@ -6,6 +6,215 @@ We assume you have already set up a bot and have Web Chat running on a page.
 
 > Sample code in this article is optimized for modern browsers. You may need to use a [transpiler](https://en.wikipedia.org/wiki/Source-to-source_compiler) (e.g. [Babel](https://babeljs.io/)) to target a broader range of browsers.
 
+## Support matrix
+
+<table>
```
Comment (corinagum) [inline SPEECH.md]: I would have preferred this as a Markdown table. :/ Oh well.

File: SPEECH.md [hunk 5]
```diff
@@ -6,6 +6,215 @@ We assume you have already set up a bot and have Web Chat running on a page.
...
+</table>
+
+### Notes
+
+1. [Web View on iOS](https://developer.apple.com/documentation/webkit/wkwebview) is not a full browser. It does not have audio recording capabilities, which is required for Cognitive Services
+2. As speech recognition is not working, speech synthesis is not tested
+3. On Safari, user gesture is always required for opening microphone. Web Chat cannot programmatically open microphone when we receive an expecting input hint
+   - https://github.com/WebAudio/web-audio-api/issues/790
+   - We are planning to unblock by changing UX, https://github.com/microsoft/BotFramework-WebChat/issues/2211
+4. Web Chat currently do not support selecting different device for audio recording
```
Comment (corinagum) [inline SPEECH.md]:
```suggestion
4. Web Chat currently does not support selecting a different device for audio recording
```

File: SPEECH.md [hunk 6]
```diff
@@ -6,6 +6,215 @@ We assume you have already set up a bot and have Web Chat running on a page.
...
+      <th>✔</th><td>4.2</td>
+      <th>❓</th><td><a href="#notes"><sup>*2</sup></a></td>
+    </tr>
+  </tbody>
+</table>
+
+### Notes
+
+1. [Web View on iOS](https://developer.apple.com/documentation/webkit/wkwebview) is not a full browser. It does not have audio recording capabilities, which is required for Cognitive Services
+2. As speech recognition is not working, speech synthesis is not tested
```
Comment (corinagum) [inline SPEECH.md]:
```suggestion
2. As speech recognition is not working (see above), speech synthesis is not tested
```
?

File: SPEECH.md [hunk 7]
```diff
@@ -6,6 +6,215 @@ We assume you have already set up a bot and have Web Chat running on a page.
...
+3. On Safari, user gesture is always required for opening microphone. Web Chat cannot programmatically open microphone when we receive an expecting input hint
```
Comment (corinagum) [inline SPEECH.md]:
```suggestion
3. On Safari, user gesture is always required for opening the microphone. Web Chat is unable to programmatically open the microphone when we receive an expecting input hint
```

File: SPEECH.md [hunk 8]
```diff
@@ -31,7 +240,7 @@ Chrome, Edge and native apps built using `WKWebView` do not support WebRTC API.
 -  Browser must [support Web Audio API][web audio api support]
    -  Available on all modern browsers on desktop and mobile platform
 
-#### Special considerations for Safari on Mac OS and iOS
+#### Special considerations for Safari on macOS and iOS
 
 Safari requires additional permission granted _implicitly_ by the user. The user needs to perform an interaction (click/tap/type) before any audio clips can be played during the browser session.
```
Comment (corinagum) [inline SPEECH.md]: wouldn't this be explicitly?

File: SPEECH.md [hunk 9]
```diff
@@ -134,6 +343,8 @@ The bot can set input hint when sending activity to the user to indicate whether
 -  `"acceptingInput"`: Web Chat will do nothing after the bot's message is spoken
 -  `"ignoringInput"`: Web Chat will explicitly close the microphone
 
+> Note: `expectingInput` is currently not working on Safari on both macOS and iOS.
```
Comment (corinagum) [inline SPEECH.md]:
```suggestion
> Note: `expectingInput` is currently not working on Safari for both macOS and iOS.
```

**Comment by compulim** [comment]
The anchors will work after it's merged.

You can try it out at https://github.com/compulim/BotFramework-WebChat/blob/feat-speech-matrix/SPEECH.md.

**Pull request merged**